tests/rl/test_health_monitor.py

```diff
@@ -0,0 +1,461 @@
+
+        metrics = monitor.env_metrics["env1"]
+        assert metrics.performance_trend == "improving"
+
+        # Create regressing trend
+        for i in range(50, 100):
+            perf = 0.8 - (i - 50) * 0.01  # Steadily declining
+            monitor.update("env2", perf, "training", "active", i)
+
+        metrics2 = monitor.env_metrics["env2"]
+        assert metrics2.performance_trend == "regressing"
+
+    def test_health_status_updates(self, monitor):
+        """Test that health status is updated based on warnings."""
+        # Start healthy
+        monitor.update("env1", 0.5, "training", "active", 0)
+        monitor.check_warnings()
+        assert monitor.env_metrics["env1"].health_status == HealthStatus.HEALTHY
+
+        # Create critical warning
+        monitor.update("env2", 0.9, "eval", "active", 100)
+        monitor.update("env2", 0.9, "eval", "graduated", 100)
+        monitor.update("env2", 0.5, "eval", "graduated", 200)  # Critical regression
+        monitor.check_warnings()
+        assert monitor.env_metrics["env2"].health_status == HealthStatus.CRITICAL
+
+    def test_health_summary(self, monitor):
+        """Test health summary generation."""
+        # Create various environments in different states
+        monitor.update("env1", 0.5, "training", "active", 10)
+        monitor.update("env2", 0.9, "eval", "active", 20)
+        monitor.update("env2", 0.9, "eval", "graduated", 20)
+        monitor.update("env3", 0.0, "training", "locked", 30)
+
+        # Create a regression
+        monitor.update("env2", 0.7, "eval", "graduated", 40)
+        monitor.check_warnings()
+
+        summary = monitor.get_health_summary()
+        assert summary["total_environments"] == 3
+        assert summary["state_distribution"]["active"] == 1
+        assert summary["state_distribution"]["graduated"] == 1
+        assert summary["state_distribution"]["locked"] == 1
+        assert summary["graduated_regressions"] == 1
+
+    def test_environment_report(self, monitor):
+        """Test detailed environment report."""
+        monitor.update("env1", 0.5, "training", "active", 10)
+        monitor.update("env1", 0.6, "eval", "active", 20)
+        monitor.env_metrics["env1"].total_training_steps = 100
+
+        report = monitor.get_environment_report("env1")
+        assert report is not None
+        assert report["env_id"] == "env1"
+        assert report["current_state"] == "active"
+        assert report["performance"]["current_train"] == 0.5
+        assert report["performance"]["current_eval"] == 0.6
+        assert report["training"]["total_steps"] == 100
+
+        # Test non-existent environment
+        assert monitor.get_environment_report("nonexistent") is None
+
+    def test_wandb_alert_rate_limiting(self, monitor):
+        """Test wandb alert rate limiting."""
+        # First alert should be allowed
+        assert monitor.should_send_wandb_alert(WarningType.GRADUATED_REGRESSION) is True
```

Comment:
it's a bit odd to expose this for testing. i'd mock the wandb call and test the behavior instead - that it sends an alert or not - that's what we care about.